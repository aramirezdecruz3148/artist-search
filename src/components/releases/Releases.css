.releases {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
  margin-left: 4vw;
}

.releases li {
  list-style: none;
}