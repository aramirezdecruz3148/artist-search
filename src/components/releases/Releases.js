import React from 'react';
import PropTypes from 'prop-types';
import Release from './Release';
import placeholderImage from '../../assets/CoverPlaceholder.png';

function Releases({ releaseArray }) {
  const releaseList = releaseArray.map(({ releaseId, releaseDate, releaseTitle, coverArtCount }) => (
    <li key={releaseId}>
      <Release 
        releaseId={coverArtCount ? `http://coverartarchive.org/release/${releaseId}/front` : placeholderImage }
        releaseDate={releaseDate}
        releaseTitle={releaseTitle}
      />
    </li>
  ));
  return (
    <ul>
      {releaseList}
    </ul>
  );
}

Releases.propTypes = {
  releaseArray: PropTypes.arrayOf(PropTypes.shape({
    releaseId: PropTypes.string.isRequired,
    releaseTitle: PropTypes.string.isRequired,
    releaseDate: PropTypes.string
  })).isRequired,
  coverArtCount: PropTypes.number.isRequired
};

export default Releases;
