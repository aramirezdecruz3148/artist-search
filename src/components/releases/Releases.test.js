import React from 'react';
import { shallow } from 'enzyme';
import Releases from './Releases';

describe('Releases component', () => {
  it('renders Releases', () => {
    const wrapper = shallow(<Releases releaseArray={[{ releaseId: '1231', releaseTitle: 'name', releaseDate: 'Jan. 31st, 1987' }, { releaseId: 'uiiuhh', releaseTitle: 'someone', releaseDate: '5/8/13' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
