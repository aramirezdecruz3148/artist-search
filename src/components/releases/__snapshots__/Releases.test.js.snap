// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Releases component renders Releases 1`] = `
<ul
  className="releases"
>
  <li
    key="1231"
  >
    <Release
      artistName="werwerw"
      releaseCover="http://coverartarchive.org/release/1231/front-250"
      releaseDate="Jan. 31st, 1987"
      releaseId="1231"
      releaseTitle="name"
    />
  </li>
  <li
    key="uiiuhh"
  >
    <Release
      artistName="werwerw"
      releaseCover="http://coverartarchive.org/release/uiiuhh/front-250"
      releaseDate="5/8/13"
      releaseId="uiiuhh"
      releaseTitle="someone"
    />
  </li>
</ul>
`;
